### ISSUE TITLE
new feature: add base32768 alphabet test to `rclone test info`

### ISSUE THREAD

**Issue by kapitainsky** [rclone/rclone#7208]
Add new flag:

`--check-base32768`

to `rclone test info`

It will create files containing every single character used by base32768 (here sample files set - https://pub.rclone.org/base32768.zip) and read them back to validate that nothing has been corrupted. It would indicate that remote is capable of using crypt with base32768 filenames' encoding.

**Comment by ncw** [comment]
OK give this a go

[v1.64.0-beta.7204.cf4035b27.fix-7208-check-base32768](https://beta.rclone.org/branch/fix-7208-check-base32768/v1.64.0-beta.7204.cf4035b27.fix-7208-check-base32768/) on branch [fix-7208-check-base32768](https://github.com/rclone/rclone/tree/fix-7208-check-base32768) (uploaded in 15-30 mins)

    rclone test info --check-base32768 remote:testdir --check-length --check-base32768

You might want to add -P or -v as the uploads take quite a while! You can increase --checkers and --transfers too.

This makes a local directory with the files to test in then syncs them to the remote and then runs check.

**Comment by kapitainsky** [comment]
wow - it was light speed.

My test:

```
 ./rclone test info --check-base32768 iDrive:test1
2023/08/04 11:04:15 NOTICE: S3 bucket test1 path test-base32768: 0 differences found
2023/08/04 11:04:15 NOTICE: S3 bucket test1 path test-base32768: 1028 matching files
// iDrive
base32768isOK = true // make sure maxFileLength for 2 byte unicode chars is the same as for 1 byte characters
```
Why only 1028 checks when alphabet is 1032?

**Comment by kapitainsky** [comment]
OK i looked at this closer:

"this alphabet contains 2**10 plus 2**2 contiguous, aligned blocks of 2**5 code points." which looking at your code"

And it is correctly 1028 - I think I counted quotes and some stray spaces to get 1032

**Comment by darthShadow** [comment]
Added to the 1.64 milestone since we have a working build.

**Comment by ncw** [comment]
Thanks for testing :-)

> OK i looked at this closer:
> 
> "this alphabet contains 2^10 plus 2^2 contiguous, aligned blocks of 2^5 code points."
> 
> And it is correctly 1028 - I think I counted quotes and some stray spaces to get 1032

I'm not sure exactly why it needs those extra 4 blocks but that is how it seems to work!

> I think it does what it should with only caveat that there is dependency in this test that local file system supports Unicode/base32768 too. Should not be a problem normally but some very edge cases.

Yes you are right. The first version didn't write the files out to disk, but then I wanted to use the sync and check primitives which run in parallel etc.

If the writing out to disk fails then rclone should error then so hopefully it will make a sensible error message!

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.64
